# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerChild.h
# Commit: ed0f9b04cb90
# Full Hash: ed0f9b04cb90ae9959ed5d37bec8454c676ff48e
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-03-07 04:32:57
# Regressor Bug: 1530223
# File Overlap Count: 1
# Description:
#   Bug 1530223: Use ThreadBound for RemoteWorkerChild members accessed on launcher thread r=perry,asuth
#   
#   This adds runtime checks that verify thread safety.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22338
# ==============================================================================

diff -r c66813164419 -r ed0f9b04cb90 dom/workers/remoteworkers/RemoteWorkerChild.h
--- a/dom/workers/remoteworkers/RemoteWorkerChild.h	Wed Mar 06 19:12:47 2019 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerChild.h	Wed Mar 06 19:00:10 2019 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_RemoteWorkerChild_h
 
 #include "mozilla/dom/PRemoteWorkerChild.h"
+#include "mozilla/ThreadBound.h"
 #include "mozilla/UniquePtr.h"
 #include "nsISupportsImpl.h"
 
@@ -99,7 +100,11 @@
   WorkerState mWorkerState;
 
   // Touched only on the owning thread (Worker Launcher).
-  nsTArray<RemoteWorkerOp> mPendingOps;
+  struct LauncherBoundData {
+    nsTArray<RemoteWorkerOp> mPendingOps;
+  };
+
+  ThreadBound<LauncherBoundData> mLauncherData;
 };
 
 }  // namespace dom
